internal/core/operations/healthcontroller/definition.go

```diff
@@ -55,11 +55,19 @@ func (def *Definition) Marshal() []byte {
 	return bytes
 }
 
-func (def *Definition) Unmarshal(raw []byte) error {
-	err := json.Unmarshal(raw, def)
+func (d *Definition) Unmarshal(raw []byte) error {
+	err := json.Unmarshal(raw, d)
 	if err != nil {
 		return fmt.Errorf("error marshalling update scheduler operation definition: %w", err)
 	}
 
 	return nil
 }
+
+func (d *Definition) NoAction() bool {
+	if d.TookAction != nil {
+		return !*d.TookAction
```

Comment:
Considering we renamed this method to HasAction, it would be more simple to read to return directly the TookAction